pci/src/vfio.rs

```diff
@@ -48,10 +48,10 @@ pub(crate) const VFIO_COMMON_ID: &str = "vfio_common";
 pub enum VfioPciError {
     #[error("Failed to create user memory region: {0}")]
     CreateUserMemoryRegion(#[source] HypervisorVmError),
-    #[error("Failed to DMA map: {0}")]
-    DmaMap(#[source] vfio_ioctls::VfioError),
-    #[error("Failed to DMA unmap: {0}")]
-    DmaUnmap(#[source] vfio_ioctls::VfioError),
+    #[error("{1}: Failed to DMA map: {0}")]
+    DmaMap(#[source] vfio_ioctls::VfioError, String),
```

Comment:
Does it not work to use the PciBdf here directly?  It implements `Display.

```suggestion
    DmaMap(#[source] vfio_ioctls::VfioError, PciBdf),
```

That way it would be more inspectable in code as well.

Reply:
> Does it not work to use the PciBdf here directly? It implements Display.
> 
> That way it would be more inspectable in code as well.

Unfortunately, it doesn't work as expected. The output seems to be `PciBdf(24)`, instead of something like `"0000:00:03.0"`.

Reply:
That's very strange — the `Display` implementation for `PciBdf` definitely returns the latter format, and the thiserror documentation suggests it should use `Display` unless you use `:?` in the format string…